spec/statement.dd

```diff
@@ -1241,15 +1241,28 @@ $(GNAME StatementNoCaseNoDefault):
     $(GLINK ScopeBlockStatement)
 )
 
-        $(P $(I Expression) is evaluated.
-        If the type of $(I Expression) is an `enum`, it is
+        $(P If the $(I IfCondition) is an *Expression*, it is evaluated.
+        If the type of the *Expression* is an `enum`, it is
         (recursively) converted to its $(GLINK2 enum, EnumBaseType).
-        Then, if the type is an integral, $(I Expression) undergoes
+        Then, if the type is an integral, the *Expression* undergoes
         $(DDSUBLINK spec/type, integer-promotions, Integer Promotions).
-        Then, the type of $(I Expression) must be either an integral or
+        Then, the type of the *Expression* must be either an integral or
         a static or dynamic array of $(CODE char), $(CODE wchar), or $(CODE dchar).
         )
 
+        $(P If an $(D auto) $(I Identifier) is provided, it is declared and
+        initialized to the value and type of the *Expression*. Its scope
+        extends from when it is initialized to the end of the *ScopeStatement*.)
+
+        $(P If a $(I TypeCtors) $(I Identifier) is provided, it is declared
+        to be of the type specified by $(I TypeCtors) and is initialized with
+        the value of the *Expression*. Its scope extends from when it is
+        initialized to the end of the *ScopeStatement*.)
+
+        $(P If a $(I Declarator) is provided, it is declared and initialized
+        to the value of the *Expression*. Its scope extends from when it is
+        initialized to the end of the *ScopeStatement*.)
```

Comment:
I think this could be more concise:
```suggestion
        $(P If an $(I Identifier) `=` prefix is provided, a variable is declared with that
        name, initialized to the value and type of the *Expression*. Its scope
        extends from when it is initialized to the end of the *ScopeStatement*.)

        $(P If $(I TypeCtors) and/or a specific type is provided for *Identifier*, those
        are used for the variable declaration which is initialized by an implicit
        conversion from the value of the *Expression*.)
```